src/map.c

```diff
@@ -620,6 +620,18 @@ int wally_keypath_xonly_public_key_verify(const unsigned char *key, size_t key_l
     return WALLY_OK;
 }
 
+int wally_merkle_path_xonly_public_key_verify(const unsigned char *key, size_t key_len,
+                                              const unsigned char *val, size_t val_len)
+{
+    struct ext_key extkey;
+
+    if (key_len != EC_XONLY_PUBLIC_KEY_LEN ||
+        keypath_key_verify(key, key_len, &extkey) != WALLY_OK ||
+        extkey.version || !val || !val_len || val_len % SHA256_LEN != 0)
```

Comment:
hmmm, with `!val_len`,  this means you can never call e.g., `wally_psbt_input_taproot_keypath_add` with a merkle length of 0.

Worked around this by adding to destination map `taproot_leaf_paths` directly via `wally_map_keypath_add`

Reply:
As discussed offline, the BIP says we have to support the root here (no hashes, i.e. 0-length), so I'll merge a fix for this to master.